plover/gui/paper_tape.py

```diff
@@ -150,6 +168,29 @@ def paper_format(self, stroke):
         text = ''.join(text)
         return text        
 
+    def _clear(self, event=None):
+        message_dialog = wx.MessageDialog(self, 'Would you like to clear the notes?', 'Clear?')
+        message_dialog.SetOKLabel("Clear")
+        if message_dialog.ShowModal() == wx.ID_OK:
+            self.listbox.Clear()
+            self.strokes.clear()
+
+    def _save(self, event=None):
+        file_name_suggestion = None
+        counter = 0
+        while not file_name_suggestion or os.path.exists(file_name_suggestion):
+            counter_string = ' (%i)' % counter if counter else ''
+            file_name_suggestion = 'steno-notes-%s%s.txt' % (time.strftime('%Y-%m-%d'), counter_string)
+            counter += 1
+        file_dialog = wx.FileDialog(self, "Save notes", "", file_name_suggestion,
+                                   "txt files (*.txt)|*", wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT)
+        if file_dialog.ShowModal() == wx.ID_OK:
+            try:
+              with open(file_dialog.GetPath(), 'w') as fd:
+                  fd.write(self.listbox.GetValue())
+            except IOError as e:
+                wx.LogError("Cannot save notes in file '%s'.\n'%s"%(file_dialog.GetPath(),str(e)))
```

Comment:
I am on linux with stumpwm. (Are you sending the notification via dbus, because I'm running plover in a sepparate dbus namespace). But I think it is better to have a dialog and specifically not a notification. If the user saves something and their saving failed, then they should be loudly told, so they don't end up thinking that they saved successfully.